### PR TITLE
feat(): async getSources

### PR DESCRIPTION
**Pull request by chyzwar**
- refactored getSources util to use aync stat and glob
- changes signature from string[] to Set<string>
- Added test setup using existing jest and tests for getSources
- Update swcrc target to avoid installing regenerator runtime
- Updated tsconfig.build.json to exclude test files and mocks

I am thinking on working on this more and slowly refactor other function yo remove any *sync functions.
Additionally, it would be interesting to run swc compilation on multiple worker threads. (unless swc already can saturate all cores?)

Can you point me with large repo using swc ? I do not have good example to test/bench my chnages.

### CODE REVIEW

**Comment by kdy1** [comment]
swc uses all cores if you use asynchronous api.


Strapi uses swc while testing

https://github.com/strapi/strapi

File: src/.swcrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "jsc": {
+        "target": "es2019",
```
Comment (kdy1) [inline src/.swcrc:3]: Doesn't this change break old node versions?

**Review by kdy1 — COMMENTED** [review]
Mostly LGTM, but I'm not sure if target is enough

File: src/.swcrc (hunk 1 above)
Comment (chyzwar) [reply src/.swcrc:3]: CLI declare node > 12.13 It should be safe to target ES2019 https://stackoverflow.com/questions/59787574/typescript-tsconfig-settings-for-node-js-12

This package is published with swc or with typescript compiled output?

**Comment by chyzwar** [comment]
I will try on strapi and try to get some meanigfull data on performance

File: src/.swcrc (hunk 1 above)
Comment (kdy1) [reply src/.swcrc:3]: Currently, it uses typescript.

Comment (chyzwar) [reply src/.swcrc:3]: tsc already target es2019.

**Comment by chyzwar** [comment]
unfortunately, strapi do not use swc/cli

**Comment by kdy1** [comment]
Hmm.. I don't know any repository using @swc/cli
The integration test of swc invokes `@swc/cli` with three.js and redux.
Maybe those can be used I think?

https://github.com/swc-project/swc/blob/master/.github/workflows/integration.yml

**Comment by chyzwar** [comment]
On current 
for three.js

current
globSources: 59.125ms
Successfully compiled 608 files with swc.
Done in 0.78s.

pr version
```
globSources: 91.64ms
Successfully compiled 608 files with swc.
Done in 0.80s.
```

I need to investigate why it slower

**Comment by chyzwar** [comment]
There are two issues
1. async glob is slower than sync version
2. Total time to compile is not reporting correctly (might not consider globbing that might be slower than whole compilation)

**Comment by chyzwar** [comment]
sorry, I missread, total time is correct. Total is 800ms when globing is 80ms

**Comment by chyzwar** [comment]
After changing glob to fast-glob I got
```
globSources: 33.037ms
Successfully compiled 608 files with swc.
Done in 0.87s.
```
https://github.com/mrmlnc/fast-glob
It 50% faster than current impl and should be backward compatible.

**Comment by kdy1** [comment]
Great! Thanks!

**Comment by kdy1** [comment]
Please ping me when the pr is ready :)

**Comment by chyzwar** [comment]
just updated now.

**Comment by kdy1** [comment]
Thanks!

**Comment by kdy1** [comment]
Reverted because of https://github.com/swc-project/cli/issues/27

**Pull request merged**